InterGalacticCurrencyConvertor/src/main/java/numerals/RomanNumerals.java

```diff
@@ -1,145 +1,154 @@
-					if (getDecimalVal(inRNums.get(j)) < vAfter){
-						++count;
-					}
-					if (count > 1) {
-						return vResult;
-					}
-					--j;
-				}
-				if (vCurRepRNCnt > 1 && !(inRNums.get(i-1).equals("i") ||
-				    inRNums.get(i-1).equals("x") || inRNums.get(i-1).equals("c") ||
-				    inRNums.get(i-1).equals("m"))) {
-					return vResult;
-				} else if (inRNums.get(i-1).equals("i") && !(inRNums.get(i).equals("v") ||
-				           inRNums.get(i).equals("x"))) {
-					return vResult;
-				} else if (inRNums.get(i-1).equals("x") && !(inRNums.get(i).equals("l") ||
-				           inRNums.get(i).equals("c"))) {
-					return vResult;
-				} else if (inRNums.get(i-1).equals("c") && !(inRNums.get(i).equals("d") ||
-				           inRNums.get(i).equals("m"))) {
-					return vResult;
-				} else if (inRNums.get(i-1).equals("v") || inRNums.get(i-1).equals("l") ||
-				           inRNums.get(i-1).equals("d")){
-					return vResult;
-				}
-				//System.out.println(inRNums.get(i-1) + " rise " + inRNums.get(i));
-			}
-			if (vRiseOcc > -1 && vRiseOcc < i && vMinimum < getDecimalVal(inRNums.get(i))) {
-				return vResult;
-			}
-			if (inRNums.get(i) != vCurRepRN) {
-				vCurRepRN = inRNums.get(i);
-				vCurRepRNCnt = 1;
-			} else {
-				++vCurRepRNCnt;
-			}
-			//System.out.println("middle");
-			if (vCurRepRNCnt > 1) {
-				if (!(inRNums.get(i).equals("i") || inRNums.get(i).equals("x") ||
-				    inRNums.get(i).equals("c") || inRNums.get(i).equals("m"))) {
-					return vResult;
-				} else if (vCurRepRNCnt == 4 && inRNums.get(i).equals(vCurRepRN)) {
-					return vResult;
-				}
-			} else if (vOccurrenceMap.get("i") > 4 || vOccurrenceMap.get("x") > 4 ||
-			           vOccurrenceMap.get("c") > 4 || vOccurrenceMap.get("m") > 4 ||
-			           vOccurrenceMap.get("v") > 1 || vOccurrenceMap.get("l") > 1 ||
-			           vOccurrenceMap.get("d") > 1) {
-				return vResult;
-			}
-
-			vBefore = vAfter;
-			if (vMinimum > getDecimalVal(inRNums.get(i))) {
-				vMinimum = getDecimalVal(inRNums.get(i));
-			}
-			//System.out.println("end " + vCurRepRNCnt);
-		}
-		vResult = true;
-		return vResult;
+	private boolean subtractionRuleFail(Character curChar,
+	                                    Character prevChar,
+	                                    int idx,
+	                                    List<Integer> rSymbolValues) {
+	  boolean isInvalid = true;
+	  int curVal = toDecimal(curChar);
+	  if (String.valueOf(prevChar).matches("[vld]")) {
+      return isInvalid;
+    } else if (prevChar.equals('i') && String.valueOf(curChar).matches("[^vx]")) {
+      return isInvalid;
+    } else if (prevChar.equals('x') && String.valueOf(curChar).matches("[^lc]")) {
+      return isInvalid;
+    } else if (prevChar.equals('c') && String.valueOf(curChar).matches("[^dm]")) {
+      return isInvalid;
+    }
+    for (int j = idx - 2; j >= 0; --j) {
+      if (rSymbolValues.get(j) < curVal) {
+        return isInvalid;
+      }
+    }
+    return !isInvalid;
```

Comment:
This is a very confusing way of returning true/false.

either do this:
boolean isInvalid = false;
for (...) {
  if (cond) {
    isInvalid = true;
    break;
  }
}

return isInvalid;

OR just do return true or false explicitly.

Reply:
changed to first way